### PR TITLE
Tokudb kill query in locktree 5.7 gca

### PR DESCRIPTION
**Pull request by vlad-lesin**
TDB-27 : Add ability to kill query that is awaiting FT locktree lock

See https://github.com/percona/percona-server/pull/1738, https://github.com/percona/PerconaFT/pull/370.

Testing:
http://jenkins.percona.com/view/5.7/job/mysql-5.7-param/934/

### CODE REVIEW

**Comment by georgelorchpercona** [comment]
This does not build for me with an out of source build using the following cmake incantation
`cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=../percona-server-install-5.7 -DMYSQL_MAINTAINER_MODE=OFF -DWITH_EMBEDDED_SERVER=OFF -DWITH_JEMALLOC=system -DWITH_BOOST=../../boost_1_59_0 -DZLIB_INCLUDE_DIR=/usr/include/ $@ ../percona-server`

File: include/my_global.h [hunk 1]
```diff
@@ -28,7 +28,9 @@
 #include "my_config.h"
 
 #define __STDC_LIMIT_MACROS	/* Enable C99 limit macros */
+#ifndef __STDC_FORMAT_MACROS
 #define __STDC_FORMAT_MACROS	/* Enable C99 printf format macros */
+#endif /* __STDC_FORMAT_MACROS */
 #define _USE_MATH_DEFINES       /* Get access to M_PI, M_E, etc. in math.h */
```
Comment (george-lorch) [inline include/my_global.h]: Why is this changed? This is in server code and will require a launchpad bug or blueprint explaining the change.

Comment (george-lorch) [reply include/my_global.h]: I see why, PerconaFT sets it in TokuSetupCompiler.cmake and that conflicts with the my_global.h definition, but I think the problem is not the server code but PerconaFT and probably best addressed there if possible. Even if allowed in my_global.h, shouldn't all __STDC definitions here get the same treatment to prevent other/future issues?

Comment (vlad-lesin) [reply include/my_global.h]: Agree, fixed in PerconaFT: portability/toku_debug_sync.h. Waiting for testing results...

**Review by george-lorch — APPROVED** [review]
Same comments re: FT submodule commit pointer as 5.6
